SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -286,78 +287,60 @@ bool FSpatialGDKEditorModule::ForEveryServerWorker(TFunction<void(const FName&,
-				for (; SpatialTestIt; ++SpatialTestIt)
-				{
-					NumberOfClients = FMath::Max(SpatialTestIt->GetNumRequiredClients(), NumberOfClients);
-				}
-				{
-					FString SnapshotForMap = ASpatialFunctionalTest::GetTakenSnapshotPath(World);
-
-					if (!SnapshotForMap.IsEmpty())
-					{
-						PIESettingsOverride.ForceUseSnapshot = SnapshotForMap;
-						// Set that we're loading from taken snapshot.
-						ASpatialFunctionalTest::SetLoadedFromTakenSnapshot();
-					}
-				}
-				break;
-			default:
-				checkf(false, TEXT("Unsupported Testing Mode"));
-				break;
-			}
-
-			PIESettingsOverride.NumberOfClients = NumberOfClients;
+			GetMutableDefault<ULevelEditorPlaySettings>()->SetSnapshotOverride(SnapshotForMap);
+			// Set that we're loading from taken snapshot.
+			ASpatialFunctionalTest::SetLoadedFromTakenSnapshot();
 		}
 	}
-	return PIESettingsOverride;
+}
+
+void FSpatialGDKEditorModule::SaveSettings() const
+{
+	// Save settings before before overriding so that they can be reverted later
+	GetMutableDefault<ULevelEditorPlaySettings>()->SaveConfig(0, *TmpLevelEditorPlaySettingsFilename);
+	GetMutableDefault<USpatialGDKSettings>()->SaveConfig(0, *TmpSpatialGDKSettingsFilename);
+	GetMutableDefault<USpatialGDKEditorSettings>()->SaveConfig(0, *TmpSpatialGDKEditorSettingsFilename);
+	GetMutableDefault<UGeneralProjectSettings>()->SaveConfig(0, *TmpGeneralProjectSettingsFilename);
+	GetMutableDefault<UEditorPerformanceSettings>()->SaveConfig(0, *TmpEditorPerformanceSettingsFilename);
+}
+
+void FSpatialGDKEditorModule::LoadSettings(const FString& TestSettingOverridesFilename)
+{
+	// Load settings from ini file which will override current settings
+	GetMutableDefault<ULevelEditorPlaySettings>()->LoadConfig(ULevelEditorPlaySettings::StaticClass(), *TestSettingOverridesFilename);
+	GetMutableDefault<USpatialGDKSettings>()->LoadConfig(USpatialGDKSettings::StaticClass(), *TestSettingOverridesFilename);
+	GetMutableDefault<USpatialGDKEditorSettings>()->LoadConfig(USpatialGDKEditorSettings::StaticClass(), *TestSettingOverridesFilename);
+	GetMutableDefault<UGeneralProjectSettings>()->LoadConfig(UGeneralProjectSettings::StaticClass(), *TestSettingOverridesFilename);
+	GetMutableDefault<UEditorPerformanceSettings>()->LoadConfig(UEditorPerformanceSettings::StaticClass(), *TestSettingOverridesFilename);
+}
+
+void FSpatialGDKEditorModule::RevertSettingsForTesting()
+{
+	// Revert settings from ini file
+	GetMutableDefault<ULevelEditorPlaySettings>()->RevertSettings(TmpLevelEditorPlaySettingsFilename);
+	GetMutableDefault<USpatialGDKSettings>()->RevertSettings(TmpSpatialGDKSettingsFilename);
+	GetMutableDefault<USpatialGDKEditorSettings>()->RevertSettings(TmpSpatialGDKEditorSettingsFilename);
+	GetMutableDefault<UGeneralProjectSettings>()->RevertSettings(TmpGeneralProjectSettingsFilename);
+	GetMutableDefault<UEditorPerformanceSettings>()->RevertSettings(TmpEditorPerformanceSettingsFilename);
 }
```

Comment:
This is a general comment across the PRs.
I think this reverting can potentially be done in another way.
Instead of saving and then reloading the configs, couldn't we save the entire object in memory and then set it back as the current settings? Not sure what the best way would be, but interestingly I see `UObject* ClassDefaultObject;` is public in `Class.h` so we could theoretically just set it to the saved instance of the settings.....

Not sure if you already tried something like this, just thought it would be worth exploring as it would save us a bunch of engine changes and future maintenance for the maps/arrays present in settings.

Reply:
Though this would pose problems if we ever want to support launching the framework in separate processes...
Alternative would be to investigate / modify the behaviour of the engine for loading empty maps / structs / arrays...

Reply:
I did try to do this in memory originally but I couldn't get it to work. I can take another look now that I have everything else working.

Reply:
I've moved the saving and reloading of configs from file to memory.